app/Controllers/LibriController.php

```
@@ -1840,9 +1851,16 @@ public function update(Request $request, Response $response, mysqli $db, int $id
             \App\Support\Hooks::do('book.save.after', [$id, $fields]);
 
-            // Gestione copie: aggiorna il numero di copie se cambiato
+            // Gestione copie: aggiorna il numero di copie se cambiato.
+            // Il conteggio di riferimento DEVE escludere le copie fuori
+            // circolazione (perso/danneggiato/manutenzione/in_restauro/
+            // in_trasferimento), perché `$fields['copie_totali']` arriva dal
+            // form pre-compilato con `libri.copie_totali`, che DataIntegrity
+            // calcola con la stessa esclusione. Con il conteggio grezzo
+            // (countByBookId) un libro con una copia fuori circolazione avrebbe
+            // current > form a ogni salvataggio e cancellerebbe una copia buona.
             $copyRepo = new \App\Models\CopyRepository($db);
```

PHPMD (2.15.0)

[error] Missing class import via use statement (line '1862', column '29'). (undefined)

(MissingImport)